streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -570,6 +575,11 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
         // so we would capture any exception and throw
         try {
             commitOffsetsOrTransaction(consumedOffsetsPerTask);
+        } catch (final TaskTimeoutExceptions taskTimeoutExceptions) {
```

Comment:
If a task is revoked, we don't need to trigger `task.timeout.ms` but can re-throw the `TimeoutException` to trigger a "dirty close" instead.